pkg/image/anaconda_ostree_installer.go

```diff
@@ -96,11 +96,13 @@ func (img *AnacondaOSTreeInstaller) InstantiateManifest(m *manifest.Manifest,
 	bootTreePipeline.UEFIVendor = img.Platform.GetUEFIVendor()
 	bootTreePipeline.ISOLabel = img.ISOLabel
 
-	ksPath := osbuild.KickstartPathOSBuild
-	if img.Kickstart != nil && img.Kickstart.Path != "" {
-		ksPath = img.Kickstart.Path
+	if img.Kickstart == nil || img.Kickstart.OSTree == nil {
+		return nil, fmt.Errorf("kickstart options not set for ostree installer")
 	}
-	bootTreePipeline.KernelOpts = []string{fmt.Sprintf("inst.stage2=hd:LABEL=%s", img.ISOLabel), fmt.Sprintf("inst.ks=hd:LABEL=%s:%s", img.ISOLabel, ksPath)}
+	if img.Kickstart.Path == "" {
+		img.Kickstart.Path = osbuild.KickstartPathOSBuild
```

Comment:
I guess for this change we want to test 
1. That it does not crash without a "kickstart" struct
2. That the default kickstart path is set

I wonder if for (2) we could amend the tests slightly, something like:
```go
diff --git a/pkg/image/installer_image_test.go b/pkg/image/installer_image_test.go
index 7fa844806..67d55e157 100644
--- a/pkg/image/installer_image_test.go
+++ b/pkg/image/installer_image_test.go
@@ -71,7 +71,8 @@ func TestContainerInstallerUnsetKSOptions(t *testing.T) {
                BIOS:       true,
                UEFIVendor: "test",
        }
-       instantiateAndSerialize(t, img, mockPackageSets(), mockContainerSpecs(), nil)
+       ser := instantiateAndSerialize(t, img, mockPackageSets(), mockContainerSpecs(), nil)
+       assert.Contains(t, ser, `"inst.ks=hd:LABEL=:/osbuild.ks"`)
 }
 
 func TestContainerInstallerUnsetKSPath(t *testing.T) {
@@ -137,7 +138,7 @@ func TestTarInstallerUnsetKSPath(t *testing.T) {
        instantiateAndSerialize(t, img, mockPackageSets(), nil, nil)
 }
 
-func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[string][]rpmmd.PackageSpec, containers map[string][]container.Spec, commits map[string][]ostree.CommitSpec) {
+func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[string][]rpmmd.PackageSpec, containers map[string][]container.Spec, commits map[string][]ostree.CommitSpec) string {
        source := rand.NewSource(int64(0))
        // math/rand is good enough in this case
        /* #nosec G404 */
@@ -147,6 +148,7 @@ func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[str
        _, err := img.InstantiateManifest(&mf, nil, &runner.CentOS{Version: 9}, rng)
        assert.NoError(t, err)
 
-       _, err = mf.Serialize(packages, containers, commits, nil)
+       ser, err := mf.Serialize(packages, containers, commits, nil)
        assert.NoError(t, err)
+       return string(ser)
 }
```
It's not super nice to just do this string match but it's probably okay.

But this does raise the question (to me, pardon my ignorance) - will this work when there is no ISOLabel set for img?

Reply:
> I guess for this change we want to test
> 
> 1. That it does not crash without a "kickstart" struct
> 2. That the default kickstart path is set
> 

1 is tested in the two test functions named `TestXInstallerUnsetKSOptions()` (the kickstart struct is never set).

> I wonder if for (2) we could amend the tests slightly, something like:
> 
> ```go
> diff --git a/pkg/image/installer_image_test.go b/pkg/image/installer_image_test.go
> index 7fa844806..67d55e157 100644
> --- a/pkg/image/installer_image_test.go
> +++ b/pkg/image/installer_image_test.go
> @@ -71,7 +71,8 @@ func TestContainerInstallerUnsetKSOptions(t *testing.T) {
>                 BIOS:       true,
>                 UEFIVendor: "test",
>         }
> -       instantiateAndSerialize(t, img, mockPackageSets(), mockContainerSpecs(), nil)
> +       ser := instantiateAndSerialize(t, img, mockPackageSets(), mockContainerSpecs(), nil)
> +       assert.Contains(t, ser, `"inst.ks=hd:LABEL=:/osbuild.ks"`)
>  }
>  
>  func TestContainerInstallerUnsetKSPath(t *testing.T) {
> @@ -137,7 +138,7 @@ func TestTarInstallerUnsetKSPath(t *testing.T) {
>         instantiateAndSerialize(t, img, mockPackageSets(), nil, nil)
>  }
>  
> -func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[string][]rpmmd.PackageSpec, containers map[string][]container.Spec, commits map[string][]ostree.CommitSpec) {
> +func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[string][]rpmmd.PackageSpec, containers map[string][]container.Spec, commits map[string][]ostree.CommitSpec) string {
>         source := rand.NewSource(int64(0))
>         // math/rand is good enough in this case
>         /* #nosec G404 */
> @@ -147,6 +148,7 @@ func instantiateAndSerialize(t *testing.T, img image.ImageKind, packages map[str
>         _, err := img.InstantiateManifest(&mf, nil, &runner.CentOS{Version: 9}, rng)
>         assert.NoError(t, err)
>  
> -       _, err = mf.Serialize(packages, containers, commits, nil)
> +       ser, err := mf.Serialize(packages, containers, commits, nil)
>         assert.NoError(t, err)
> +       return string(ser)
>  }
> ```
> 
> It's not super nice to just do this string match but it's probably okay.
> 
> But this does raise the question (to me, pardon my ignorance) - will this work when there is no ISOLabel set for img?

It "works" in the sense that it creates a manifest, but perhaps it shouldn't because it sets the `isolabel` option in the `grub2.iso` stage to an empty string, which is "valid" only in the strictest sense (i.e. it will pass schema validation, but probably break in the build or boot).

For both the isolabel and the default kickstart path, I keep flip-flopping on whether we should test them separately (and fix them up) vs waiting for the big changes coming with otk integration (https://github.com/osbuild/images/pull/764#issuecomment-2189611407).  Looking around the installer pipelines, I think there's some kickstart path handling that's buggy, so I should at least test for that now.

Reply:
Handled in follow-up (#781).